src/components/PopupWithForm.js

```diff
@@ -4,41 +4,54 @@ import Popup from "./Popup.js";
 
 //this is a child of the popup class
 export default class PopupWithForm extends Popup {
-  constructor({modalSelector, handleFormSubmit}) {
+  constructor({
+    modalSelector,
+    handleFormSubmit
+  }) {
     super({
       modalSelector
     }); // to pass the popup to parent
 
     this._modalForm = this._modalElement.querySelector('.modal__form') // now we go access to this thanks to passing it thru the super()
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._modalForm.querySelectorAll('.modal__input')
+    this._submitButton = this._modalElement.querySelector('.modal__button')
+    //i could get the default button text from the above variable and use it afterwards for the loading text function
   }
-  _getinputValues(){
+  _getinputValues() {
     const formValues = {}
-    this._inputList.forEach((input)=>{
-    formValues[input.name] = input.value;
-    //sets key of that property equals to the name of each form
+    this._inputList.forEach((input) => {
+      formValues[input.name] = input.value; // for each input it adds a key value pair with
+      //sets key of that property equals to the name of each form
+      // you are passing the input values from the form
     })
     //which collects data from all the input fields and returns it as an object. This data should then be passed to the submission handler as an argument.
-   return formValues;
+    return formValues;
   }
-  setEventListeners(){
+  setEventListeners() {
 
     //class should add a submit event listener to the form and call the setEventListeners() method of the parent class.
     //this for the submit listenters
-    this._modalForm.addEventListener('submit',(evt)=>{
+    this._modalForm.addEventListener('submit', (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getinputValues());
 
     })
-     super.setEventListeners();
+    super.setEventListeners();
   }
 
   close() {
     this._modalForm.reset()
     super.close();
   }
 
+  setLoadingButtonText(isLoading, loadingText = "Saving..."){
+    if(isLoading){
+     return this._submitButton.textContent = loadingText;
+    }
+    this._submitButton.textContent = "Save"
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```